ios/Exponent/Versioned/Core/Api/EXNotifications.m

```diff
@@ -92,41 +93,124 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(__unused RCTPromiseRejectBlock)reject)
 {
-  UILocalNotification *notification = [self _localNotificationFromPayload:payload];
+  UNMutableNotificationContent* content = [self _localNotificationFromPayload:payload];
 
   [EXUtil performSynchronouslyOnMainThread:^{
-    [RCTSharedApplication() presentLocalNotificationNow:notification];
+    UNNotificationRequest* request = [UNNotificationRequest
+                                      requestWithIdentifier:content.userInfo[@"id"] content:content trigger:nil];
+    [[UNUserNotificationCenter currentNotificationCenter] addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
+      if (error != nil) {
+        reject(@"E_NOTIF", [NSString stringWithFormat:@"Could not add a notification request: %@", error.localizedDescription], error);
+      } else {
+        resolve(content.userInfo[@"id"]);
+      }
+    }];
   }];
 
-  resolve(notification.userInfo[@"id"]);
+}
+
+RCT_EXPORT_METHOD(addCategory: (NSString *)categoryId
+                  actions: (NSArray *)actions
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(__unused RCTPromiseRejectBlock)reject)
+{
+  NSMutableArray<UNNotificationAction *> * actionsArray = [[NSMutableArray alloc] init];
+  
+  for(NSArray *action in actions) {
+    int optionsInt = [(NSNumber *)action[2] intValue];
```

Comment:
The action parsing needs to be more robust to prevent runtime crashes and to communicate what each field means. Create a small struct that has five fields and write a "parser" that creates a struct from the array, and sets an `NSError **` if parsing fails. Then we can reject this promise if there is an error.

Reply:
BUMP